src/base/NekRSProblem.C

```diff
@@ -828,4 +919,45 @@ NekRSProblem::addExternalVariables()
   }
 }
 
+void
+NekRSProblem::calculateMeshVelocity(int e, const field::NekWriteEnum & field)
+{
+  int len = _nek_mesh->order() + 2;
+  double *tmp_meshvel = (double *)calloc(len*len, sizeof(double));
+  double dt = _timestepper->getCurrentDT();
+
+  switch (field)
+  {
+    case field::mesh_velocity_x:
+      for (int i=0; i <len * len; i++)
+      {
+        tmp_meshvel[i] = ( _displacement_x[i]-_nek_mesh->prev_disp_x().at((e*len*len) + i))
+                           /dt/_U_ref;
+      }
+      writeBoundarySolution(e, field::mesh_velocity_x,tmp_meshvel);
+      _nek_mesh->updateDisplacement(e, _displacement_x, field::x_displacement);
+      break;
+    case field::mesh_velocity_y:
+      for (int i=0; i <len * len; i++)
+      {
+        tmp_meshvel[i] = ( _displacement_y[i]-_nek_mesh->prev_disp_y().at((e*len*len) + i))
+                           /dt/_U_ref;
+      }
+      writeBoundarySolution(e, field::mesh_velocity_y,tmp_meshvel);
+      _nek_mesh->updateDisplacement(e, _displacement_y, field::y_displacement);
+      break;
+    case field::mesh_velocity_z:
+      for (int i=0; i <len * len; i++)
+      {
+        tmp_meshvel[i] = ( _displacement_z[i]-_nek_mesh->prev_disp_z().at((e*len*len) + i))
+                           /dt/_U_ref;
+      }
+      writeBoundarySolution(e, field::mesh_velocity_z,tmp_meshvel);
+      _nek_mesh->updateDisplacement(e, _displacement_z, field::z_displacement);
+      break;
+    default:
+      throw std::runtime_error("Unhandled NekWriteEnum in NekRSProblem::interpolateBoundaryDisplacement!\n");
```

Comment:
Please use `mooseError` - it basically wraps this type of command, but is meant to work with the rest of MOOSE's console system.